shpkpr/exceptions.py

```diff
@@ -37,6 +37,8 @@ def wrapper(*args, **kwargs):
                 if six.PY2:
                     six.reraise(exception_to_rewrap_with, ei.message, tb)
                 else:
-                    six.reraise(exception_to_rewrap_with, exception_to_rewrap_with(*ei.args), tb)
+                    ei_args = ei.args if ei.args else (ei.message,) if hasattr(ei, "message") else None
+                    ei_args = ei_args if ei_args is not None else (str(ei),)
```

Comment:
NAB: Too much logic here. Seems noisy. I get why it is done like that but there might be a nicer/easier way? But it obv works so I don't think its a massive deal.